src/Sylius/Behat/Service/NotificationChecker.php

```diff
@@ -66,29 +67,47 @@ public function checkEditionNotification($resource)
     {
         $message = sprintf('%s has been successfully updated.', $this->humanizeResourceName($resource));
 
-        $this->checkSuccessNotificationMessage($message);
+        $this->checkNotification($message, NotificationType::success());
     }
 
     /**
-     * @param string $message
-     *
-     * @throws NotificationExpectationMismatchException
+     * {@inheritdoc}
      */
-    public function checkSuccessNotificationMessage($message)
+    public function checkNotification($message, NotificationType $type)
     {
-        if ($this->notificationAccessor->hasSuccessMessage() && $this->notificationAccessor->hasMessage($message)) {
+        if ($this->hasType($type) && $this->hasMessage($message)) {
             return;
         }
 
         throw new NotificationExpectationMismatchException(
-            'success',
+            $type,
             $message,
-            $this->notificationAccessor->getMessageType(),
+            $this->notificationAccessor->getType(),
             $this->notificationAccessor->getMessage()
         );
     }
 
     /**
+     * @param NotificationType $type
+     *
+     * @return bool
+     */
+    private function hasType(NotificationType $type)
+    {
+        return (string) $type === (string) $this->notificationAccessor->getType();
```

Comment:
I'm just predicting the WTF moments happening if someone tries to compare types without casting to a string :water_buffalo: